### PR TITLE
New logo

### PR DESCRIPTION
**Pull request by zhuiks**
Logo files from #214 discussion
- png and ico are 256x256
- icns is converted from 512x512

Let me know if we need other sizes

Maybe it can be a part of changing product name #171

### CODE REVIEW

**Comment by tobias-klein** [comment]
I like the logo. It doesn't look so nice in the scaled down version on Windows, though. Windows is the primary user group. Do you see a possibility to optimize a bit more?


image

**Comment by alerque** [comment]
I see what you mean about looking bad in that Windows screen shot, but ironically the screenshot in the issue scaled to the width of the GitHub issues column looks great. In other words it's not just the size that's a problem as it scales much smaller just fine, it's something specifically unlucky about the rasterization size and settings being used and it can surely be fixed ... by hand hinting if we have to but by design and export tweaks too.

**Comment by alerque** [comment]
I suggest also having some sort of build rules around so that the icons can be regenerated from the SVG. This tooling wouldn't need to be part of the every day build procedure since the raster resources are also in the repository but at least a reproducible chain of rules to get the output again if something were to change would be handy.

**Comment by zhuiks** [comment]
I can tweak the width of the scroll outline. I notice that a bit thicker width looks better on the small sizes. 

But I agree that the main problem is auto scaling. I just figured out that different icon sizes (16x16, 32x32, 48x48, and 256x256) can be (should be) combined to one ico file. In the previous commit I just converted 256x256 png to ico. So I'll try combining

**Comment by zhuiks** [comment]
@tobias-klein how about now?
I used imagemagick `convert`: 
```
convert icons/ezra.png -define icon:auto-resize="256,48,32,16" icons/ezra.ico
```
If we use it in build script for Windows in github actions does it need to be installed?

How about MacOS how new logo is looking there?

**Comment by tobias-klein** [comment]
Thanks for your hard work on this! I'll review the updates tomorrow!

**Comment by alerque** [comment]
We should be able get Image Magick running in the CI builders just fine. I would note that the `convert` interface from v6 and prior is deprecated in favor of `magick` in v7+. I suggest something in `build_scripts` to regenerate all the resources that are derived from the SVG. It could even be setup so each platform only generated the relevant resources, although it will take a little tweaking to get the right workflow on all systems. At the very least having it scripted in a way developers can refresh the resources will make future tweaks easier. Whether or not the resulting PNGs should even be tracked in the repo is up for debate, for now I say leave them until such a time as they are built per-platform.

**Comment by tobias-klein** [comment]
This is how it looks on macOS:

**DMG Installer**
Bildschirmfoto 2021-04-14 um 19 15 51

**Dock icon**
Bildschirmfoto 2021-04-14 um 19 18 48

**Comment by tobias-klein** [comment]
** Update on Windows**
image

image

**Comment by tobias-klein** [comment]
Does the Mac version intentionally have a black background?

**Comment by zhuiks** [comment]
> Does the Mac version intentionally have a black background?

No it should be transparent, I converted it with [cloud converter](https://cloudconvert.com/png-converter)
Let me try again

**Comment by zhuiks** [comment]
according to [this](https://www.codeproject.com/articles/1067619/what-icon-sizes-are-used-by-windows) standard sizes for Windows are 16, 32, 48, 256 (must have) and additional are 20, 24, 40, 60, 72
```bash
convert icons/windows-source/*.png icons/ezra.ico
```

according to [this](https://gist.github.com/zlbruce/883605a635df8d5964bab11ed75e46ad) and playing with `png2icns` for MacOS sizes are 16, 32, 128, 256, 512, 1024
```bash
sudo apt install icnsutils
png2icns icons/ezra.icns icons/mac-os-source/*.png
```

**Comment by zhuiks** [comment]
I don't have any access to Windows or MacOS computer now. You have to test it. Sorry, guys

**Comment by tobias-klein** [comment]
> I don't have any access to Windows or MacOS computer now. You have to test it. Sorry, guys

Something for the weekend! :) No worries.

**Comment by tobias-klein** [comment]
**Android** 
image

image

It seems like the lower edge could need some anti aliasing.
On Android the PNG version is used.

**Comment by zhuiks** [comment]
> It seems like the lower edge could need some anti aliasing.
On Android the PNG version is used.

Yes, I'm not satisfied. Is it from the latest commit? How do you build for Android?

**Comment by zhuiks** [comment]
I got a hold of the Windows, result is still similar in the for the window header. Although app icon of the same size looks nice:

ezra-windows-screenshot

**Comment by tobias-klein** [comment]
The icon in the top left corner of the window was already looking mediocre before ... (with the old logo). We may need to accept that for the time being. Maybe other open source Electron apps have solved this somehow, could be interesting to have a look at some other apps. I don't think it's a blocking issue for merging this pull request. Let's iterate!
Are you still looking into any other enhancements regarding the icon? Otherwise I would consider to merge this and have it released in **0.18.0**.

**Comment by tobias-klein** [comment]
The icon you saw on Android was from the latest icons/ezra.png.

**Comment by zhuiks** [comment]
> Are you still looking into any other enhancements regarding the icon? Otherwise I would consider to merge this and have it released in 0.18.0.

I'll research a bit about that little corner icon. And I'll try to make logo nicer on Android. 

Let me know how the latest logo looks on MacOS

**Comment by tobias-klein** [comment]
> Let me know how the latest logo looks on MacOS

Bildschirmfoto 2021-04-17 um 08 43 23

Pretty good I would say! Note that I still need to change the name to Ezra Bible App here.

**Comment by tobias-klein** [comment]
Latest update on Android:

image

I feel like the inner rectangle is not necessary. I'd prefer to have the icon a bit bigger. What do you think?

**Comment by zhuiks** [comment]
What version of Cordova are we using?

Which version of Android/launcher are you testing with?

Seems like icons for Android are depending on the launcher.

**Comment by tobias-klein** [comment]
Cordova 7
Default Android Launcher

I am testing on two devices, one running Android 10, the other Android 7.
Why not a newer Cordova version? Because this is what nodejs-mobile requires. See https://code.janeasystems.com/nodejs-mobile/getting-started-cordova

**Comment by zhuiks** [comment]
I followed this in `cofig.xml`:
https://cordova.apache.org/docs/en/7.x/config_ref/images.html#android

Newer versions of Android support adaptive icons with two foreground/background layers. Which was also generated by default with Android studio's asset manager.

But I think we are good with the round one.

**Comment by tobias-klein** [comment]
The adaptive icons are not yet supported by Cordova 7. It seems like Cordova 9 is required for that (see https://github.com/ionic-team/cordova-res/issues/34). Anyway ... I just checked once more on Android and the issue about the missing icon in the app switcher was simply a caching problem. When I cleanly regenerated the app with the latest sources everything looked fine (using the round icon). Therefore I think we are ready to merge now! Thanks for your hard work on this @zhuiks! This is awesome :+1:

**Review by tobias-klein — APPROVED** [review]
